api/src/main/java/uk/gov/dhsc/htbhf/claimant/reporting/payload/ReportPropertiesFactory.java

```diff
@@ -0,0 +1,105 @@
+        eventPropertiesMap.put(HIT_TYPE_KEY, HIT_TYPE_VALUE);
+        eventPropertiesMap.put(EVENT_CATEGORY.getFieldName(), CLAIM.name());
+        eventPropertiesMap.put(EVENT_ACTION.getFieldName(), context.getClaimAction().name());
+        eventPropertiesMap.put(EVENT_VALUE.getFieldName(), 0);
+        eventPropertiesMap.put(QUEUE_TIME.getFieldName(), calculateQueueTime(context.getTimestamp()));
+        eventPropertiesMap.put(CUSTOMER_ID.getFieldName(), context.getClaim().getId());
+        return eventPropertiesMap;
+    }
+
+    private Map<String, Object> createCustomDimensionMap(ReportClaimMessageContext context) {
+        Map<String, Object> customDimensions = new TreeMap<>();
+        customDimensions.put(USER_TYPE.getFieldName(), ONLINE.name());
+        ClaimantCategory claimantCategory = claimantCategoryCalculator
+                .determineClaimantCategory(context.getClaim().getClaimant(), context.getDatesOfBirthOfChildren(), context.getTimestamp().toLocalDate());
+        customDimensions.put(CLAIMANT_CATEGORY.getFieldName(), claimantCategory.getDescription());
+        PostcodeData postcodeData = context.getClaim().getPostcodeData();
+        customDimensions.put(LOCAL_AUTHORITY.getFieldName(), postcodeData.getAdminDistrict());
+        customDimensions.put(LOCAL_AUTHORITY_CODE.getFieldName(), postcodeData.getCodes().getAdminDistrict());
+        customDimensions.put(COUNTRY.getFieldName(), postcodeData.getCountry());
+        customDimensions.put(POSTCODE_OUTCODE.getFieldName(), postcodeData.getOutcode());
+        customDimensions.put(WESTMINSTER_PARLIAMENTARY_CONSTITUENCY.getFieldName(), postcodeData.getParliamentaryConstituency());
+        customDimensions.put(CLINICAL_COMMISSIONING_GROUP.getFieldName(), postcodeData.getCcg());
+        customDimensions.put(CLINICAL_COMMISSIONING_GROUP_CODE.getFieldName(), postcodeData.getCodes().getCcg());
+        return customDimensions;
+    }
+
+    private Map<String, Object> createCustomMetricMap(ReportClaimMessageContext context) {
+        Map<String, Object> customMetrics = new TreeMap<>();
+        Claimant claimant = context.getClaim().getClaimant();
+        customMetrics.put(CLAIMANT_AGE.getFieldName(), Period.between(claimant.getDateOfBirth(), LocalDate.now()).getYears());
+        long childrenUnder4 = context.getDatesOfBirthOfChildren().stream().filter(dob -> dob.isAfter(LocalDate.now().minusYears(4))).count();
+        long childrenUnder1 = context.getDatesOfBirthOfChildren().stream().filter(dob -> dob.isAfter(LocalDate.now().minusYears(1))).count();
+        customMetrics.put(CHILDREN_UNDER_ONE.getFieldName(), childrenUnder1);
+        customMetrics.put(CHILDREN_BETWEEN_ONE_AND_FOUR.getFieldName(), childrenUnder4 - childrenUnder1);
+        LocalDate expectedDeliveryDate = claimant.getExpectedDeliveryDate();
+        if (expectedDeliveryDate == null || expectedDeliveryDate.isBefore(LocalDate.now())) {
```

Comment:
Should this be using the same component that determines if a delivery date is considered pregnant or not?

Reply:
Agreed. Will add in next PR along with moving some other refactors.